grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1093,6 +1093,19 @@ class StockMovementService {
             throw new IllegalArgumentException("Could not find shipment for stock movement with ID ${stockMovement.id}")
         }
 
+        if (stockMovement.requisition.status == RequisitionStatus.ISSUED) {
+            shipment.name = stockMovement.name != shipment.name || shipment.destination == stockMovement.destination ? stockMovement.name : stockMovement.generateName()
+
+            if (shipment.destination != stockMovement.destination) {
+                shipment.outgoingTransactions?.each { transaction ->
+                    transaction.destination = stockMovement.destination
```

Comment:
@jmiranda can I update the transactions like that?

Reply:
@pmuchowski I don't like it, but I guess it should work. I haven't read the ticket, but is this for the use case where a user changes the destination after issuing the stock movement?

Reply:
By the way, there are two transactions for every shipped shipment. Outbound transaction and inbound transaction. Do we need to update both?

Reply:
@jmiranda yes, it's used when the destination is changed.
I looked into the code and inbound transaction is created when shipment is received and the destination cannot be changed at this point (on the SM edit page).